### ISSUE TITLE
`gutter` option for `table.hline()`

### ISSUE THREAD

**Issue by bjoluc** [typst/typst#4743]
### Description

In good old booktabs manner, I would love to see a `gutter` (or the like) option for `table.hline()`, so `table.hline()` can introduce an additional vertical space around the line.

I'm currently achieving something similar with a rather hacky workaround:

```typst
#let hline(columns) = {
  return (
    table.cell(colspan: columns, inset: 1.5pt)[],
    table.hline(),
    table.cell(colspan: columns, inset: 1.5pt)[],
  )
}
```

so I can replace `table.hline()` by `..hline(4)` (if the table has 4 columns) to add vertical space around the line.

### Use Case

The booktabs LaTeX package has some spacing around horizontal lines (`\midrule`) within the table by default. A `gutter` option for `table.hline()` would allow template authors to style tables similar to booktabs in their templates (by `set`ting a default value for the `gutter` parameter). :slightly_smiling_face:

**Comment by bluebear94** [comment]
How would `table.hline`’s gutter interact with the table’s `gutter` setting?

**Comment by bjoluc** [comment]
@bluebear94 Good point! Intuitively, I would assume they would add up. At least for the booktabs-style usage of `table.hline()`, that would be quite convenient. Alternatively, maybe the larger gutter value of both should be used? WDYT?

**Comment by bluebear94** [comment]
I agree that adding the values would make the most sense.
